warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java

```diff
@@ -1825,4 +1948,52 @@ public void testIgnoreIrrelevantZones() {
         // Ensure that we get the hit if we evaluate null zone
         Assert.assertEquals(Collections.singleton(zone2.getZone()), ContentFunctions.phrase((Object) null, termOffsetMap, terms));
     }
+    
+    /**
+     * Verify that if gatherPhraseOffsets is false, that even when there is a phrase index for matching excerpt field, it is not recorded.
+     */
+    @Test
+    public void testGatherPhraseOffsetsIsFalse() {
+        String query = buildFunction(ContentFunctions.CONTENT_WITHIN_FUNCTION_NAME, "1", Constants.TERM_OFFSET_MAP_JEXL_VARIABLE_NAME, "'dog'", "'cat'");
+        Expression expr = engine.createExpression(query);
+        
+        List<TermWeightPosition> list1, list2;
+        list1 = asList(1);
+        list2 = asList(2);
+        
+        termOffSetMap.putTermFrequencyList("dog", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), list1)));
+        termOffSetMap.putTermFrequencyList("cat", new TermFrequencyList(Maps.immutableEntry(new Zone("CONTENT", true, eventId), list2)));
+        termOffSetMap.setGatherPhraseOffsets(false);
+        termOffSetMap.setExcerptFields(Collections.singleton("CONTENT"));
+        
+        context.set(Constants.TERM_OFFSET_MAP_JEXL_VARIABLE_NAME, termOffSetMap);
+        Object o = expr.evaluate(context);
+        
+        Assert.assertTrue(expect(o, true));
+        assertPhraseOffsetsEmpty();
+    }
+    
+    /**
+     * Verify that if gatherPhraseOffsets is true, if a phrase index for is found for a non-excerpt field, it is not recorded.
+     */
+    @Test
+    public void testNonMatchingExcerptFields() {
```

Comment:
An alternate test for this would be an event with a phase for two different fields (BODY and CONTENT), excerpts enabled, and excerpt fields set to only one field. Not required for approval, just another edge case to consider.

Reply:
This was a long time coming, but I added a test to cover this edge case.

Reply:
Perfect, thank you! Apologies for the lapse in attention